server/lib/notifications/agents/webpush.ts

```diff
@@ -0,0 +1,205 @@
+          requestId: payload.request?.id,
+          actionUrl: `/${payload.media?.mediaType}/${payload.media?.tmdbId}`,
+        };
+      case Notification.MEDIA_DECLINED:
+        return {
+          notificationType: Notification[type],
+          subject: payload.subject,
+          message: 'Your request was declined.',
+          image: payload.image,
+          mediaType: payload.media?.mediaType,
+          tmdbId: payload.media?.tmdbId,
+          requestId: payload.request?.id,
+          actionUrl: `/${payload.media?.mediaType}/${payload.media?.tmdbId}`,
+        };
+      case Notification.MEDIA_FAILED:
+        return {
+          notificationType: Notification[type],
+          subject: payload.subject,
+          message: 'The request failed.',
+          image: payload.image,
+          mediaType: payload.media?.mediaType,
+          tmdbId: payload.media?.tmdbId,
+          requestId: payload.request?.id,
+          actionUrl: `/${payload.media?.mediaType}/${payload.media?.tmdbId}`,
+        };
+      case Notification.MEDIA_PENDING:
+        return {
+          notificationType: Notification[type],
+          subject: payload.subject,
+          message: `New ${
+            payload.media?.mediaType === MediaType.MOVIE ? 'movie' : 'series'
+          } request pending approval.\nRequested by: ${
+            payload.request?.requestedBy.displayName
+          }`,
+          image: payload.image,
+          mediaType: payload.media?.mediaType,
+          tmdbId: payload.media?.tmdbId,
+          requestId: payload.request?.id,
+          actionUrl: `/${payload.media?.mediaType}/${payload.media?.tmdbId}`,
+        };
+    }
+  }
+
+  public shouldSend(type: Notification): boolean {
+    if (
+      this.getSettings().enabled &&
+      hasNotificationType(type, this.getSettings().types)
+    ) {
+      return true;
+    }
+
+    return false;
+  }
+
+  public async send(
+    type: Notification,
+    payload: NotificationPayload
+  ): Promise<boolean> {
+    logger.debug('Sending web push notification', { label: 'Notifications' });
+    const userRepository = getRepository(User);
+    const userPushSubRepository = getRepository(UserPushSubscription);
+    const settings = getSettings();
+
+    let pushSubs: UserPushSubscription[] = [];
+
+    const mainUser = await userRepository.findOne({ where: { id: 1 } });
+
+    if (payload.notifyUser) {
+      const notifySubs = await userPushSubRepository.find({
+        where: { user: payload.notifyUser.id },
+      });
+
+      pushSubs = notifySubs;
+    } else {
+      const users = await userRepository.find();
+
+      const manageUsers = users.filter((user) =>
+        user.hasPermission(Permission.MANAGE_REQUESTS)
+      );
+
+      const allSubs = await userPushSubRepository
+        .createQueryBuilder('pushSub')
+        .where('pushSub.userId IN (:users)', {
+          users: manageUsers.map((user) => user.id),
+        })
+        .getMany();
+
+      pushSubs = allSubs;
+    }
```

Comment:
We also need to verify that the notification recipient has enabled this notification agent in their user settings.